src/config.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Limited
+# See LICENSE file for licensing details.
+
+"""Spark History Server configuration."""
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from constants import (
+    CONFIG_KEY_S3_ACCESS_KEY,
+    CONFIG_KEY_S3_CREDS_PROVIDER,
+    CONFIG_KEY_S3_ENDPOINT,
+    CONFIG_KEY_S3_LOGS_DIR,
+    CONFIG_KEY_S3_SECRET_KEY,
+)
+from utils import WithLogging
+
+
+class AbstractSparkHistoryServerConfig(WithLogging, ABC):
+    """Abstraction for Spark History Server Configuration."""
+
+    def __init__(self):
+        self.config = {}
+
+    @abstractmethod
+    def get(self) -> Dict[str, str]:
+        """Get Config."""
+        pass
+
+    @abstractmethod
+    def populate(self, conf: Dict[str, str]) -> None:
+        """Populate Config From Model or Relation etc."""
+        pass
```

Comment:
**suggestion** instead of this UX:

```
config = SparkHistoryServerConfig()
config.populate(self.model.config)
```

what about using the constructor directly 

```
config = SparkHistoryServerConfig(self.model.config)
```

In this case the constructor would just be 

```
def __init__(self, config: Dict[str, Any]):
    self.config = config
```

and we could avoid the `get` (as that would just be `spark_config.config`) and `populate` methods

But maybe I'm missing the underlying reason for having the abstract class and have the implementation of multiples populate and get.